jetson/nav/utilities.cpp

```diff
@@ -3,165 +3,135 @@
-Odometry createOdom( const Odometry & current, double bearing, const double distance, Rover * rover )
-{
-    bearing = degreeToRadian( bearing );
-    double latChange = distance * cos( bearing ) * LAT_METER_IN_MINUTES;
-    double lonChange = distance * sin( bearing  ) * rover->longMeterInMinutes();
-    Odometry newOdom = addMinToDegrees( current, latChange, lonChange );
+Odometry createOdom(const Odometry& current, double bearing, const double distance, shared_ptr<Rover> rover) {
+    bearing = degreeToRadian(bearing);
+    double latChange = distance * cos(bearing) * LAT_METER_IN_MINUTES;
+    double lonChange = distance * sin(bearing) * rover->longMeterInMinutes();
+    Odometry newOdom = addMinToDegrees(current, latChange, lonChange);
     return newOdom;
 }
 
 // Caclulates the bearing between the current odometry and the
 // destination odometry.
-double calcBearing( const Odometry& start, const Odometry& dest )
-{
-    double currentLat = degreeToRadian( start.latitude_deg, start.latitude_min );
-    double currentLon = degreeToRadian( start.longitude_deg, start.longitude_min );
-    double destLat = degreeToRadian( dest.latitude_deg, dest.latitude_min );
-    double destLon = degreeToRadian( dest.longitude_deg, dest.longitude_min );
-
-    double verticleComponentDist = EARTH_RADIUS * sin( destLat - currentLat );
-    double noneuclidDist = estimateNoneuclid( start, dest );
-
-    double bearing = acos( verticleComponentDist / noneuclidDist );
-    if( currentLon > destLon )
-    {
+double calcBearing(const Odometry& start, const Odometry& dest) {
+    double currentLat = degreeToRadian(start.latitude_deg, start.latitude_min);
+    double currentLon = degreeToRadian(start.longitude_deg, start.longitude_min);
+    double destLat = degreeToRadian(dest.latitude_deg, dest.latitude_min);
+    double destLon = degreeToRadian(dest.longitude_deg, dest.longitude_min);
+
+    double verticleComponentDist = EARTH_RADIUS * sin(destLat - currentLat);
+    double noneuclidDist = estimateNoneuclid(start, dest);
+
+    double bearing = acos(verticleComponentDist / noneuclidDist);
+    if (currentLon > destLon) {
         bearing = 2 * PI - bearing;
     }
 
-    if( verticleComponentDist < 0.001 && verticleComponentDist > -0.001 )
-    {
-        if( currentLon < destLon )
-        {
+    if (verticleComponentDist < 0.001 && verticleComponentDist > -0.001) {
+        if (currentLon < destLon) {
             bearing = PI / 2;
-        }
-        else
-        {
+        } else {
             bearing = 3 * PI / 2;
         }
     }
-    return radianToDegree( bearing );
+    return radianToDegree(bearing);
 } // calcBearing()
 
 // // Calculates the modulo of degree with the given modulus.
-double mod( const double degree, const int modulus )
-{
-    double mod = fmod( degree, modulus );
-    if( mod < 0 )
-    {
-        return ( mod + modulus );
-    }
-    return mod;
+double mod(const double degree, const int modulus) {
```

Comment:
you removed the const for this in the header file

Reply:
https://stackoverflow.com/questions/51353648/const-qualification-of-parameters-in-function-declaration